src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1425,3 +1428,13 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     this.logger.info(s);
   }
 }
+
+export function isMediaDeviceInfo(device: unknown): device is MediaDeviceInfo {
+  return (
+    typeof device === 'object' &&
+    'deviceId' in device &&
+    'groupId' in device &&
+    'kind' in device &&
+    'label' in device
```

Comment:
Are these four fields always present in all `MediaDeviceInfo` instances?

Reply:
Verified in Chrome, FF, Safari, Safari iOS. All four fields are present (although groupId = "" in Safari).